File: tests/assets/rules/semgrep/best_practices/flag_secure.java
public class FlagSecureTestActivity extends Activity {
  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    // ruleid:android_prevent_screenshot
    getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
                         WindowManager.LayoutParams.FLAG_SECURE);

    setContentView(R.layout.main);
  }
}


if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.HONEYCOMB) {
    // ruleid:android_prevent_screenshot
    getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
}



        final Activity activity = getCurrentActivity();

        if (activity != null) {
            activity.runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    // ruleid:android_prevent_screenshot
                    activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
                }
            });
        }

File: tests/assets/rules/semgrep/best_practices/kotlin/flag_secure.kt

fun protect(window: android.view.Window) {
    // ruleid:android_prevent_screenshot
    window.setFlags(
        WindowManager.LayoutParams.FLAG_SECURE,
        WindowManager.LayoutParams.FLAG_SECURE,
    )
    // ruleid:android_prevent_screenshot
    window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
}
